components/VueLayerMarker.vue

```diff
@@ -24,7 +37,54 @@ export default defineComponent({
         src: {
             type: String,
             default: "/marker.png"
+        },
+        selectSrc: {
+            type: String,
+            default: "/marker.png"
+        },
+        details: { // ToDo - define positionData object like index.vue (or re-use that one by referencing it, somehow)
+          type: Object,
+          default: () => {}
         }
+    },
+    emits: ["popup-toggled"],
+    data () {
+      return {
+        iconSrc: this.src,
+        selectediconSrc: this.selectSrc
+      };
+    },
+    methods: {
+      onSelect (e : any) {
+        /*
+         * WARNING: the needed coordinates are regular long/lat (°N °E)
+         * BE SURE TO CONVERT
+         */
+        e.feature.getGeometry().transform("EPSG:3857", "EPSG:4326");
+        const markerCoords : Array<number> = e.feature.getGeometry().getCoordinates();
+        // TODO - properly document in our docs
+        /*
+         * WARNING: the object that calls the event is NOT necesarily the correct component for the marker we clicked on
+         * due to this behaviour, we cannot just pass this.details. We use the properties stored in the vl-feature of the component
+         * that was clicked, and extract the details needed. This might also need updating in the future.
+         *
+         * We blame this behaviour on VueLayers being quirky.
+         */
+        const f = e.feature.getProperties();
+        // convert timestamp to readable format
+        const timestamp : Date = new Date(f.timestamp * 1000);
+        const tsString : string = timestamp.toLocaleString();
+        const details = { id: f.id, longitude: f.longitude, latitude: f.latitude, timestamp: tsString };
+        this.$root.$emit("popup-toggled", markerCoords, details);
+        /*
+         * console.log(this.displayDetails, this.details, this.coordinates);
+         * re-transform to avoid disappearing markers
+         */
+        e.feature.getGeometry().transform("EPSG:4326", "EPSG:3857");
```

Comment:
Using suggestion from previous comment should make this unnecessary.

Reply:
Indeed it does